sdk/core/azure-core/src/main/java/com/azure/core/util/polling/implementation/PollResult.java

```diff
@@ -41,8 +41,10 @@ public PollResult setStatus(String status) {
             this.status = LongRunningOperationStatus.SUCCESSFULLY_COMPLETED;
         } else if (PollingConstants.STATUS_FAILED.equalsIgnoreCase(status)) {
             this.status = LongRunningOperationStatus.FAILED;
+        } else if (PollingConstants.STATUS_CANCELLED.equalsIgnoreCase(status)) {
+            this.status = LongRunningOperationStatus.USER_CANCELLED;
         } else {
-            this.status = LongRunningOperationStatus.fromString(status, true);
+            this.status = LongRunningOperationStatus.fromString(status, false);
```

Comment:
Other status would be `isComplete=false`

Reply:
This is a behavior breaking change but it's the right change to make as we should terminate the poller on only known terminal states. Should we consider introducing a compat switch here?

Reply:
I have another string to add here.

[ForkJoinPool-1-worker-5] INFO  - {"az.sdk.message":"HTTP response","contentLength":"88","statusCode":200,"url":"REDACTED","durationMs":315,"Content-Length":"88","Content-Type":"application/json","retry-after":"6","x-ms-request-id":"REDACTED","x-envoy-upstream-service-time":"REDACTED","OpenAI-Processing-Ms":"REDACTED","x-ms-client-request-id":"e7de899a-dd5a-478f-9243-b29136b2b3e4","apim-request-id":"REDACTED","Strict-Transport-Security":"REDACTED","x-content-type-options":"REDACTED","x-ms-region":"REDACTED","Date":"Fri, 14 Jul 2023 15:37:24 GMT","body":"{\"created\":1689349043,\"id\":\"1bc96f44-1fa6-4ee4-980e-422195dd7fa1\",**\"status\":\"notRunning\"**}"}

Reply:
@mssfang 

I guess it is OpenAI? That is where we find the problem of LRO not working properly, and hence this fix.

Reply:
Got you. I was trying to merge for last month, but the behavior change has certain impact and Srikanta recommend Jonathan to review as well.

I will try to get it in this month.

Reply:
If we were to have a compat switch, I'd like to also have a plan for how long it would hang around before we removed it, and also that we held the switch close to our chest so that it does not needlessly become relied upon.

Reply:
Spoke to @JonathanGiles offline and decided not have the compact switch as this is not an issue for most services and OpenAI (where we found this issue) will not have LROs for image generation.